Added missing dependencies to WF embedded configurations; added test

MatousJobanek opened the pull request:
Added missing dependencies into WF embedded configurations:
WF8 
`org.jboss.remotingjmx:remoting-jmx:2.0.1.Final`
`org.jboss.logging:jboss-logging:3.2.1.Final`
WF9
`org.jboss.remotingjmx:remoting-jmx:2.0.1.Final`
into the containers.yaml file. Without these dependencies the containers failed.

All of the WF embedded containers requires a dependency `org.jboss.logmanager:jboss-logmanager`  and the property `java.util.logging.manager` has to be set to `org.jboss.logmanager.LogManager`. I added the dependency to the list of dependencies to be transitively fetched into a developer's project (not sure if it is correct approach or not). The only thing that developer has to do to make the container running is to set the property.

Added also test execution configurations into the `all` profile

bartoszmajsak commented:
Thanks for the fix @MatousJobanek!

> All of the WF embedded containers requires a dependency `org.jboss.logmanager:jboss-logmanager` and the property `java.util.logging.manager` has to be set to `org.jboss.logmanager.LogManager`. I added the dependency to the list of dependencies to be transitively fetched into a developer's project (not sure if it is correct approach or not). The only thing that developer has to do to make the container running is to set the property.

I think this dependency should be fetched the same way as the others, right? Otherwise it's always available disregards what kind of container you are using. As for the property - can we automatically set it before starting the container? Or at least log the warning to inform the user. Obviously we should document it somewhere too :)

MatousJobanek commented:
As for the dependency: yeah, that was my first idea, but it's not working in this way (if I put it into the containers.yaml) - I don't know why. 
As for the property: I tried it. I tried to put System.setProperty(...) here [TargetController.start](https://github.com/arquillian/arquillian-container-chameleon/blob/master/src/main/java/org/arquillian/container/chameleon/controller/TargetController.java#L84) as well as here [TargetController.setup](https://github.com/arquillian/arquillian-container-chameleon/blob/master/src/main/java/org/arquillian/container/chameleon/controller/TargetController.java#L73) and it doesn't work. To be honest, I haven't tried to set the property through the ContainerConfiguration

bartoszmajsak commented:
Maybe we should try with `ContainerConfiguration` and see if that solves the problem. But frankly, it's strange because even if we run in the forked JVM (using surefire for example) those operations should be invoked there. 

As for the dependency - this is a bit confusing - maybe worth deeper troubleshooting. As an temporary solution we could move it to respected profiles instead

MatousJobanek commented:
The reason why the dependency is needed in the pom.xml file is that the `org.jboss.logmanager.LogManager` is set in `maven-surefire-plugin` - in other words, it is required when a test case is being started. If the dependency is not set in the pom.xml file, then CNFE is thrown before running the actual test:
http://pastebin.test.redhat.com/426121
When the dependency is defined,but the configuration in surefire plugin is missing, then this RE is thrown when the container is being started:
http://pastebin.test.redhat.com/426122

This was the explanation why the dependency has to be in pom.xml file. Now, how to hack the settings to not having it in maven-surefire-plugin ....?

MatousJobanek commented:
So I tried many hacks and approaches, but without any success. 
I moved the `jboss-logmanager` dependency to the `all` profile (to keep it testable) and added a note containing information about this necessary additional settings into the README file

bartoszmajsak commented:
LGTM
